x-pack/legacy/plugins/epm/server/registry/index.ts

```diff
@@ -86,6 +86,13 @@ export function pathParts(path: string): AssetParts {
     [pkgkey, service, type, file] = path.replace(`dataset/${dataset}/`, '').split('/');
   }
 
+  // This is to cover for the fields.yml files inside the "fields" directory
+  if (file === undefined) {
```

Comment:
Can you add some confirmation that we have something reasonable? e.g. `&& type.endsWith('.yml')` or something

Reply:
added. I think we need to rethink / cleanup this part later on as I think both dataset and fields are a bit "hacky", but didn't want to spend time on it now.

Reply:
👍 I didn't see this comment when I made approval comment, but totally agree.